sklearn/metrics/_classification.py

```diff
@@ -2013,63 +2038,110 @@ class after being classified as negative. This is the case when the
         positive_likelihood_ratio = np.nan
         negative_likelihood_ratio = np.nan
-    else:
-        tn, fp, fn, tp = cm.ravel()
-        support_pos = tp + fn
-        support_neg = tn + fp
-        pos_num = tp * support_neg
-        pos_denom = fp * support_pos
-        neg_num = fn * support_neg
-        neg_denom = tn * support_pos
-
-        # If zero division warn and set scores to nan, else divide
-        if support_pos == 0:
-            msg = "no samples of the positive class were present in the testing set "
-            if raise_warning:
-                warnings.warn(msg, UserWarning, stacklevel=2)
-            positive_likelihood_ratio = np.nan
-            negative_likelihood_ratio = np.nan
-        if fp == 0:
+
+    if fp == 0:
+        if zero_division == "warn":
             if tp == 0:
-                msg = "no samples predicted for the positive class"
+                msg_beginning = (
+                    "No samples were predicted for the positive class and "
+                    "`positive_likelihood_ratio` is "
+                )
             else:
-                msg = "positive_likelihood_ratio ill-defined and being set to nan "
-            if raise_warning:
-                warnings.warn(msg, UserWarning, stacklevel=2)
+                msg_beginning = "`positive_likelihood_ratio` is ill-defined and "
+            msg_end = "set to np.nan. Use the `zero_division` param to control this "
+            "behavior."
+            warnings.warn(msg_beginning + msg_end, UndefinedMetricWarning, stacklevel=2)
             positive_likelihood_ratio = np.nan
-        else:
-            positive_likelihood_ratio = pos_num / pos_denom
-        if tn == 0:
-            msg = "negative_likelihood_ratio ill-defined and being set to nan "
-            if raise_warning:
-                warnings.warn(msg, UserWarning, stacklevel=2)
+        elif (
+            isinstance(zero_division.get("LR+", None), (int, float))
+            and zero_division.get("LR+", None) == 1
+        ):
+            positive_likelihood_ratio = np.float64(zero_division["LR+"])
+        elif np.isinf(zero_division.get("LR+", None)):
+            positive_likelihood_ratio = np.inf
+        else:  # np.isnan(zero_division["LR+"])
+            positive_likelihood_ratio = np.nan
+    else:
+        positive_likelihood_ratio = pos_num / pos_denom
+
+    if tn == 0:
+        if zero_division == "warn":
+            msg = (
+                "`negative_likelihood_ratio` is ill-defined and set to np.nan. "
+                "Use the `zero_division` param to control this behavior."
+            )
+            warnings.warn(msg, UndefinedMetricWarning, stacklevel=2)
             negative_likelihood_ratio = np.nan
-        else:
-            negative_likelihood_ratio = neg_num / neg_denom
+        elif isinstance(
+            zero_division.get("LR-", None), (int, float)
+        ) and zero_division.get("LR-", None) in [0, 1]:
```

Comment:
Here, I don't think that we will raise an error if we don't have any other thing that 0/1. I'm thinking that we could always create when possible a dictionary and check that the values in the dict are the one expected (0/1/np.inf). This would be safer at this stage.

Reply:
This is very interesting. Before I jump into making this, let me reformulate to clarify if we are talking about the same idea:

Should we accept any value that the metrics could output as a valid input in the the `zero_division` dict?
Should we allow it for `class_likelihood_ratios` only or also for the other classification metrics? 
Would it be useful to someone?

I feel I could not judge if it was useful and would rather stick with how `zero_division` works on the other metrics by implementing it in a similar way here. (Which means restricting possible values to the min and max scores.) If you or someone goes "yes, yes yes" for the above questions, I'd be happy to learn more about this idea and to make it work this way.

Reply:
After we had talked about this, I have now implemented a lenient check of the input values for the `zero_division` param at the beginning of the function, that allows all the values, the ratios could take.